### PR TITLE
add additional resource regarding for loops to fundamentals 4

### PR DESCRIPTION
**Pull request by JohnKeysCloud**
## Because
I noticed a lot of people had trouble figuring out how for loops work. This article breaks down how they work while solving a problem that can be found in the tasks section of https://javascript.info/while-for within the Fundamentals 4 lesson.

## This PR
- add [https://medium.com/@cyclonestudios/loop-ception-breaking-down-nested-for-loops-in-javascript-e3387ca1d6fc](https://github.com/TheOdinProject/curriculum/pull/url) link to additional resources section

## Issue
n/a

## Additional Information
n/a

## Pull Request Requirements
-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: foundations/javascript_basics/fundamentals-4.md [hunk 1]
```diff
@@ -51,7 +51,7 @@ Follow the steps below to get started. Once you complete Step 1, be sure to use
 
 This section contains helpful links to related content. It isn’t required, so consider it supplemental.
 
-* It looks like this lesson doesn't have any additional resources yet. Help us expand this section by contributing to our curriculum.
+- [Breaking Down JavaScript for Loops - Medium](https://medium.com/@cyclonestudios/loop-ception-breaking-down-nested-for-loops-in-javascript-e3387ca1d6fc)
```
Comment (thatblindgeye) [inline foundations/javascript_basics/fundamentals-4.md:54]:
```suggestion
- If you're confused about how nested `for` loops work, [Breaking down nested for loops in JavaScript](https://medium.com/@cyclonestudios/loop-ception-breaking-down-nested-for-loops-in-javascript-e3387ca1d6fc) can help clear things up. If you did the tasks from the "Loops" section above, the example here should look familiar!
```

Or something along those lines, just to kind of tell users what this link would be good for outside of the link title.

**Review by thatblindgeye — CHANGES_REQUESTED** [review]

File: foundations/javascript_basics/fundamentals-4.md (hunk 1 above)
Comment (JohnKeysCloud) [reply foundations/javascript_basics/fundamentals-4.md:54]: I agree. That works!

**Comment by thatblindgeye** [comment]
So after reading through the article more closely, personally I'm wondering if providing a resource just for nested loops. While they may be used, I don't believe are really seen outside the one practice exercise in that `javascript.info` link, and it could be something that users toy around with to figure out. Has there been a significant number of users on the Discord expressing confusion over the topic?

 Will leave this open as it can be mulled over some more.

**Comment by JohnKeysCloud** [comment]
Some found the resource useful not only for the exercise in the javscript.info link, but also to solidify their understanding of for-loops in general.

**Comment by thatblindgeye** [comment]
Will be closing this for now since it is more of a limited additional resource that really only focuses on a single exercise from the lesson's assignment. The assigned reading on for loops go into a good amount of detail already.

Don't feel discouraged in suggesting other resources in the future. We don't add every resource, and it really depends on the context, but we're always open to adding or updating stuff in the curriculum.

**Pull request closed without merging**